### PR TITLE
[PORTS] Dragon & Carp changes

### PR DESCRIPTION
**Pull request by Therealdoooc213**
## About The Pull Request

Increases carp health, updates dragon sprite, makes invalid carp rift locations a balloon message, and gives dragons night vision.

Ports:
1. https://github.com/tgstation/tgstation/pull/83811 (partially)

## Why It's Good For The Game

DRAGON CHANGES

Security being able to cut out all of the lights and then just kill you unless you manage to light them on fire is silly.
Having the only way to tell you cant deploy a carp rift being tiny red let in the bottom right of your screen sucks, the only other way to tell is by checking your objectives.

CARP CHANGES
Ive increased the health of Advanced Space Carp to 60, previously they'd instantly die in 2 laser shots, and the 500 second respawn timer made them even more of a pushover, they also do more object damage.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

(the dragon sprites in this video are no longer in)
https://github.com/user-attachments/assets/2e9cc8a5-f824-4d3f-8321-409ac15e5f9b
image

</details>

## Changelog
:cl: mc-oofert, toastgoats, therealdoooc
add: Trying to put down a rift as a space dragon in a invalid area now shows a balloon alert.
balance: Dragons now have night vision.
balance: Carp have more health and deal more object damage.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Carps definitely needed more health considering the low spawn rate

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 1]
```diff
@@ -243,10 +243,10 @@
 /mob/living/simple_animal/hostile/carp/advanced
 	name = "advanced space carp"
 	desc = "A ferocious, fang-bearing creature that resembles a fish."
-	maxHealth = 40
-	health = 40
+	maxHealth = 60
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]: This increases (40 to 60 and 15 to 20) are so minor that they probably aren't going to make any real impact.

Comment (Therealdoooc213) [reply code/modules/mob/living/simple_animal/hostile/carp.dm]: Fair enough, I think 80 and 25 would be better.

**Comment by ClownMoff** [comment]
Our dragon sprite is so much better and iconic we don't need the /TG/ sprite

**Comment by Therealdoooc213** [comment]
> Yea, I hate the TG dragon sprite. Is it actually changed in this PR since the testing evidence doesn't reflect that. The current sprite is a mix between both and was especially made due to how unfitting the TG sprite was when it was originally ported.
> 
> Also they should get proper night vision that lets them see through byond darkness and not just making the lighting plane more transparent.

Fair enough, will change it back!

**Comment by MarkusLarsson421** [comment]
Can you give carps partly nightvision?

File: code/modules/antagonists/space_dragon/carp_rift.dm [hunk 2]
```diff
@@ -24,7 +24,7 @@
 		return
 	var/area/A = get_area(S)
 	if(!(A in dragon.chosen_rift_areas))
-		to_chat(S, "<span class='warning'>You can't summon a rift here!</span>")
+		owner.balloon_alert(owner, "can't summon a rift here! check your objectives!")
```
Comment (PowerfulBacon) [inline code/modules/antagonists/space_dragon/carp_rift.dm]:
```suggestion
		owner.balloon_alert(owner, "can't summon a rift here, check your objectives!")
```

**Comment by Therealdoooc213** [comment]
> Also they should get proper night vision that lets them see through byond darkness and not just making the lighting plane more transparent.

Unsure how to do this. i tried `see_in_darkness = 100` but it didnt work.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**